src/common/meta/middleware_data.rs

```diff
@@ -109,13 +113,18 @@ impl RumExtraData {
 
             user_agent_hashmap.insert("ip".into(), ip_address.into());
 
-            let ip: IpAddr = match ip_address.split(':').next().unwrap().parse() {
-                Ok(ip) => ip,
-                Err(e) => {
-                    log::error!("Error parsing IP address: {}, {}", ip_address, e);
-                    IpAddr::V4(std::net::Ipv4Addr::new(127, 0, 0, 1))
-                }
-            };
+            let ip = ip_address
+                .parse::<IpAddr>()
+                .or_else(|_| {
+                    ip_address
+                        .parse::<SocketAddr>()
+                        .map(|sock_addr| sock_addr.ip())
+                        .map_err(|e| {
+                            log::error!("Error parsing IP address: {}, {}", &ip_address, e);
+                            e
+                        })
+                })
+                .unwrap_or_else(|_| IpAddr::V4(std::net::Ipv4Addr::new(127, 0, 0, 1)));
```

Comment:
Maybe we can split this part into a function, then the unit test will be easier.

Reply:
Sure, can put this as a utils function.